### PR TITLE
mgr/dashboard: add TLS

### PR DESCRIPTION
**Pull request by jcsp**
Hunted down the specific thing that was breaking in CherryPy 3.5.0 (which is old, but it's what's in Xenial and Fedora 27), and added a monkey patch for it.

This is DNM because the standby mode is still just http:// -- writing this made me realise that I neglected to enable standby modules to access `get_store`, so this can't get at its certificates.  Didn't notice this when updating the restful module because it doesn't implement standby mode.

### CODE REVIEW

**Comment by rjfd** [comment]
@jcsp tested in python 3 and it's working!

File: src/pybind/mgr/dashboard/module.py [hunk 1]
```diff
@@ -7,16 +7,39 @@
...
     from urllib.parse import urljoin
+
 try:
     import cherrypy
 except ImportError:
     # To be picked up and reported by .can_run()
     cherrypy = None
 
+
+if cherrypy is not None and cherrypy.__version__ == "3.5.0":
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/module.py]: Do you know what version it was fixed in? I wonder if checking for 3.5.0 only is too narrow? How about "if version is lower than <version we know this was fixed in>"?

**Comment by sebastian-philipp** [comment]
Can you also add pyopenssl to the requirements.txt?
```patch
diff --git a/src/pybind/mgr/dashboard/requirements.txt b/src/pybind/mgr/dashboard/requirements.txt
index fc623ade31..e71ca3fd0b 100644
--- a/src/pybind/mgr/dashboard/requirements.txt
+++ b/src/pybind/mgr/dashboard/requirements.txt
@@ -23,6 +23,7 @@ pylint==1.8.2
 pytest==3.3.2
 pytest-cov==2.5.1
 pytz==2017.3
+pyopenssl=17.5.0
 requests==2.18.4
 Routes==2.4.1
 singledispatch==3.4.0.3
```

File: src/pybind/mgr/dashboard/module.py (hunk 1 above)
Comment (jcsp) [reply src/pybind/mgr/dashboard/module.py]: Turns out it was fixed in 3.7.0, so I've updated this to be a range (think it's very unlikely anyone's going to put 3.6.0 in the wild but people are crazy so who knows!)

**Comment by tchaikov** [comment]
```
************* Module dashboard.module
E: 30, 0: No name 'version' in module 'distutils' (no-name-in-module)
E: 30, 0: Unable to import 'distutils.version' (import-error)
E: 36, 8: No name 'wsgiserver' in module 'cherrypy' (no-name-in-module)
E: 36, 8: Unable to import 'cherrypy.wsgiserver.wsgiserver2' (import-error)
C: 48, 0: Import "from mgr_module import MgrModule, MgrStandbyModule" should be placed at the top of the module (wrong-import-position)
W:127,12: Attribute 'cert_tmp' defined outside __init__ (attribute-defined-outside-init)
W:136,12: Attribute 'pkey_tmp' defined outside __init__ (attribute-defined-outside-init)
C: 30, 0: standard import "from distutils.version import StrictVersion" should be placed before "from OpenSSL import crypto" (wrong-import-order)
```

might want to cleanup the lint warnings.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-27_07:23:02-rados-wip-kefu-testing-2018-04-27-0902-distro-basic-smithi/

**Comment by tchaikov** [comment]
```
py27-lint runtests: commands[1] | pycodestyle --max-line-length=100 --exclude=.tox,venv,frontend --ignore=E402,E121,E123,E126,E226,E24,E704,W503 .
./module.py:41:9: E301 expected 1 blank line, found 0
./module.py:198:1: E302 expected 2 blank lines, found 1
./module.py:329:28: E741 ambiguous variable name 'O'
ERROR: InvocationError: '/home/jenkins-build/build/workspace/ceph-pull-requests/build/py27-lint/bin/pycodestyle --max-line-length=100 --exclude=.tox,venv,frontend --ignore=E402,E121,E123,E126,E226,E24,E704,W503 .'
```

@jcsp ^

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-28_01:51:03-rados-wip-kefu-testing-2018-04-28-0125-distro-basic-smithi/

**Comment by tchaikov** [comment]
retest this please.

**Comment by jcsp** [comment]
Woohoo, linter is happy now (after silencing the E741 check, which is bogus in my opinion -- I agree with the person at https://github.com/PyCQA/pycodestyle/issues/700)

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-05-01_16:39:19-rados-wip-kefu-testing-2018-05-01-2255-distro-basic-smithi/

http://tracker.ceph.com/issues/23961

**Pull request merged**